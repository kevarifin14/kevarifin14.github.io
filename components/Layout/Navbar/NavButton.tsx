import { ReactNode } from 'react';

type NavButtonProps = {
  onClick: () => void,
  children: ReactNode,
}

export default function NavButton({ onClick, children }: NavButtonProps) {
  return (
    <button
      type="button"
      className="bg-white rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none"
      onClick={onClick}
    >
      {children}
    </button>
  );
}
