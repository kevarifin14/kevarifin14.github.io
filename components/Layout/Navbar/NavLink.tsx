import Link from 'next/link';
import { HiMenu } from 'react-icons/hi';

export interface ILink {
  href: string,
  title: string,
  icon?: typeof HiMenu,
}

type NavLinkProps = {
  link: ILink,
  showIcon?: boolean,
}

export default function NavLink({ link, showIcon = false }: NavLinkProps) {
  return (
    <Link href={link.href}>
      <a className="flex items-center space-x-3 text-lg font-medium border-transparent hover:bg-transparent">
        {showIcon && (
          <div>
            <link.icon className="flex-shrink-0 h-6 w-6 text-primary" />
          </div>
        )}
        <p>{link.title}</p>
      </a>
    </Link>
  );
}
