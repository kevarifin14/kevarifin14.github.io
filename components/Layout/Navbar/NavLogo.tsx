import Image from 'next/image';
import Link from 'next/link';

export default function NavLogo() {
  return (
    <div className="relative h-16 w-16 cursor-pointer">
      <Link href="/">
        <Image
          src="/blue.svg"
          layout="fill"
          objectFit="contain"
        />
      </Link>
    </div>
  );
}
