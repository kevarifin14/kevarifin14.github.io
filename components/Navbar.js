import { FaBars } from 'react-icons/fa';

import Link from 'next/link';
import styles from 'styles/Navbar.module.scss';

function Navbar({ showLogo }) {
  return (
    <div className={styles.nav}>
      <span style={{ opacity: showLogo ? 1 : 0 }}>
        <Link href="/">
          <div style={{ display: 'flex', alignItems: 'center', cursor: 'pointer' }}>
            <img src="/blue.svg" height={40} style={{ marginBottom: '1em', marginTop: '1em' }} alt="logo" />
            <h2 style={{ whiteSpace: 'nowrap', marginBottom: '1em', marginTop: '1em' }}>Kevin Arifin </h2>
          </div>
        </Link>
      </span>

      <label htmlFor={styles.navToggle} className={styles.burgerMenu}>
        <FaBars />
      </label>
      <input type="checkbox" id={styles.navToggle} />
      <div className={styles.rightMenu}>
        <Link href="/blog"><a className={styles.navLink}>Blog</a></Link>
        <Link href="/tb"><a className={styles.navLink}>Newsletter</a></Link>
        <Link href="/about"><a className={styles.navLink}>About</a></Link>
      </div>

    </div>
  );
}

export default Navbar;
