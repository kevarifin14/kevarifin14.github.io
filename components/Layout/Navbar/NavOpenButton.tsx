import { HiOutlineMenu } from 'react-icons/hi';

import NavButton from './NavButton';

type NavOpenButtonProps = {
  onClick: () => void,
}

export default function NavOpenButton({ onClick }: NavOpenButtonProps) {
  return (
    <NavButton onClick={onClick}>
      <HiOutlineMenu className="h-6 w-6" />
    </NavButton>
  );
}
