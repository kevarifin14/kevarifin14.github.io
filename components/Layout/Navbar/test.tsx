import renderer from 'react-test-renderer';

import Navbar from '.';

describe('Navbar', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Navbar />);
    expect(tree).toMatchSnapshot();
  });
});
