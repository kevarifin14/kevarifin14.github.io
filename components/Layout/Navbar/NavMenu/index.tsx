import NavLink from './NavLink';

type NavMenuProps = {
  links: any,
  className: string,
}

export default function NavMenu({ links, className }: NavMenuProps) {
  const navMenuClassName = [
    'space-x-10',
    className,
  ].join(' ');

  return (
    <nav className={navMenuClassName}>
      {links.map(({ title, href }) => <NavLink key={href} href={href}>{title}</NavLink>)}
    </nav>
  );
}
