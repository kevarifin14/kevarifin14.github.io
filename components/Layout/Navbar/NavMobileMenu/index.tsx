import NavLink from '../NavLink';
import NavLogo from '../NavLogo';
import NavCloseButton from './NavCloseButton';

export default function NavMobileMenu({ links, onClick }) {
  return (
    <div className="rounded-lg shadow-lg ring-1 ring-black ring-opacity-5 bg-white divide-y-2 divide-gray-50 m-4">
      <div className="pt-5 pb-6 px-5">

        <div className="flex items-center justify-between">

          <NavLogo />

          <div className="-mr-2">
            <NavCloseButton onClick={onClick} />
          </div>
        </div>

        <div className="mt-6">
          <nav className="grid gap-y-8">
            {links.map((link) => (
              <button onClick={onClick}>
                <NavLink link={link} showIcon />
              </button>
            ))}
          </nav>
        </div>

      </div>
    </div>
  );
}
