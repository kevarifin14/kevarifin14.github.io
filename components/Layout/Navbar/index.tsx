import { Transition } from '@headlessui/react';
import { useRouter } from 'next/router';
import { useState } from 'react';
import {
  HiBookOpen, HiDocumentText, HiMail, HiUser,
} from 'react-icons/hi';

import { ILink } from './NavLink';
import NavLogo from './NavLogo';
import NavMenu from './NavMenu';
import NavMobileMenu from './NavMobileMenu';
import NavMenuButton from './NavOpenButton';

export default function Navbar() {
  const router = useRouter();
  const [isVisible, setIsVisible] = useState(false);

  const links: ILink[] = [
    {
      title: 'Blog',
      href: '/blog',
      active: router.pathname.includes('/blog'),
      icon: HiDocumentText,
    },
    {
      title: 'Newsletter',
      href: '/newsletter',
      active: router.pathname.includes('/newsletter'),
      icon: HiMail,
    },
    {
      title: 'About',
      href: '/about',
      active: router.pathname.includes('/about'),
      icon: HiUser,
    },
  ];

  return (
    <div className="relative max-w-7xl mx-auto w-full px-4 sm:px-6 lg:px-8">
      <div className="flex justify-between items-center pt-4 md:space-x-10">

        <NavLogo />

        <div className="-mr-2 -my-2 md:hidden">
          <NavMenuButton onClick={() => setIsVisible(true)} />
        </div>

        <NavMenu className="hidden md:flex" links={links} />

      </div>

      <Transition
        show={isVisible}
        enter="duration-200 ease-out"
        enterFrom="opacity-0 scale-95"
        enterTo="opacity-100 scale-100"
        leave="duration-100 ease-in"
        leaveFrom="opacity-100 scale-100"
        leaveTo="opacity-0 scale-95"
        className="absolute top-0 inset-x-0 py-2 transition transform origin-top-right md:hidden z-50"
      >
        <NavMobileMenu
          links={links}
          onClick={() => setIsVisible(false)}
        />
      </Transition>
    </div>
  );
}
