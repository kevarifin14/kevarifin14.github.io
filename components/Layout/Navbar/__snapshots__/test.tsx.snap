// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<div
  className="relative max-w-7xl mx-auto w-full"
>
  <div>
    <div
      className="flex justify-between items-center border-b-2 border-gray-100 py-4 md:space-x-10"
    >
      <div
        className="relative h-16 w-16 cursor-pointer"
      >
        <div
          style={
            Object {
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "left": 0,
              "margin": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <img
            decoding="async"
            onClick={[Function]}
            onMouseEnter={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "contain",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "visibility": "hidden",
                "width": 0,
              }
            }
          />
        </div>
      </div>
      <div
        className="-mr-2 -my-2 md:hidden"
      >
        <button
          className="bg-white rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="h-6 w-6"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <nav
        className="space-x-10 hidden md:flex"
      >
        <a
          className="text-lg font-medium text-gray-500 hover:text-gray-900 no-underline"
          href="/blog"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Blog
        </a>
        <a
          className="text-lg font-medium text-gray-500 hover:text-gray-900 no-underline"
          href="/tb"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Newsletter
        </a>
        <a
          className="text-lg font-medium text-gray-500 hover:text-gray-900 no-underline"
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          About
        </a>
      </nav>
    </div>
  </div>
</div>
`;
